container-images/scripts/rag_framework

```diff
@@ -33,6 +35,22 @@ def eprint(e, exit_code):
 # Helper Classes and Functions
 
 
+def wait_for_llama_server(base_url: str, timeout: int = 20):
+    start = time.time()
+    with httpx.Client() as client:
+        while time.time() - start < timeout:
+            try:
+                # Try a health endpoint or root path; adjust if your llama-server has a different path
+                resp = client.get(f"{base_url}/health", timeout=5)
+                if resp.status_code == 200:
+                    return True
+            except httpx.RequestError:
+                pass  # Can't connect yet, server is not up.
+            time.sleep(2)
```

Comment:
Is this sleep here really necessary? The GET request would block (for 5s) when the server doesn't reply and when it replies, the /health endpoint should be ready (soon).